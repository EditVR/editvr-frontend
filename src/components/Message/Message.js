/**
 * @file Message.js
 * Exports a component that renders a given form message.
 */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Typography, withStyles } from '@material-ui/core';

import MessageStyles from './Message.style';

const Message = ({ children, classes, type }) =>
  children ? (
    <Typography
      variant="subheading"
      className={classNames(classes.message, {
        [classes.error]: type === 'error',
        [classes.info]: type === 'info'
      })}
    >
      {children}
    </Typography>
  ) : null;

Message.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.shape({
    message: PropTypes.string.isRequired,
    error: PropTypes.string.isRequired,
    info: PropTypes.string.isRequired
  }).isRequired,
  type: PropTypes.oneOf(['error', 'info'])
};

Message.defaultProps = {
  children: null,
  type: 'info'
};

export default withStyles(MessageStyles)(Message);
