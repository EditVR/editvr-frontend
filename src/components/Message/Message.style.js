/**
 * @file Message.style.js
 * Exports Message component styles.
 */

import red from '@material-ui/core/colors/red';

export default theme => ({
  message: {
    marginTop: theme.spacing.unit
  },
  error: {
    color: red[500]
  },
  info: {
    color: 'rgba(255, 255, 255, 0.7)'
  }
});
