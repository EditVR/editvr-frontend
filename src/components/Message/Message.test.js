/**
 * @file Message.test.js
 * Contains tests for Message.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Message from './Message';

describe('<Message />', () => {
  it('Matches its snapshot', () => {
    expect(
      renderer
        .create(<Message>Error: your input is borked lol</Message>)
        .toJSON()
    ).toMatchSnapshot();
  });
});
