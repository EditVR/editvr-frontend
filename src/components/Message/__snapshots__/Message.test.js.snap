// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> Can render an error message 1`] = `
<h3
  className="MuiTypography-root-4 MuiTypography-subheading-11 Message-message-1 Message-error-2"
>
  Error: your input is borked lol
</h3>
`;

exports[`<Message /> Can render an info message 1`] = `
<h3
  className="MuiTypography-root-4 MuiTypography-subheading-11 Message-message-1 Message-info-3"
>
  Hey: you might want to know about this lol
</h3>
`;

exports[`<Message /> Matches its snapshot 1`] = `
<h3
  className="MuiTypography-root-4 MuiTypography-subheading-11 Message-message-1 Message-info-3"
>
  Error: your input is borked lol
</h3>
`;
